stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -1567,4 +1567,5 @@ function lyap(A::StridedMatrix{T}, C::StridedMatrix{T}) where {T<:BlasFloat}
     rmul!(Q*(Y * adjoint(Q)), inv(scale))
 end
 lyap(A::StridedMatrix{T}, C::StridedMatrix{T}) where {T<:Integer} = lyap(float(A), float(C))
+lyap(A::Adjoint{T,Matrix{T}}, C::Adjoint{T,Matrix{T}}) where T = lyap(Matrix(A), Matrix(C))
```

Comment:
Why not generalize this to be an `AbstractMatrix` for both `A` and `C`:
```
lyap(A::AbstractMatrix{T}, C::AbstractMatrix{T}) where T = lyap(Matrix(A), Matrix(C))
```

That will allow for the adjoints to be passed in, as well as other matrix types to also be used (such as `Tridiagonal`, etc.) and make this bare-bones example code work (it will not currently work because it doesn't allow the `Tridiagonal` dispatch on the second argument).

```julia
A = randn(5,5)
B = randn(5,5)

lyap( A, Tridiagonal( B'*B ) )
```

Reply:
This should be
```julia
function lyap(A::AbstractMatrix, C::AbstractMatrix)
    T = promote_type(float(eltype(A)), float(eltype(C)))
    return lyap(copy_to_array(A, T), copy_to_array(C, T))
end
```
This will make sure that the matrices are strided and have float eltype, in one go.